pandas/tests/indexing/test_floats.py

```diff
@@ -534,102 +541,94 @@ def test_slice_integer_frame_getitem(self, index_func):
+            self.check(result, s, indexer, False)
 
             # positional indexing
             msg = (
                 "cannot do slice indexing "
-                r"on {klass} with these indexers \[-10\.0\] of "
+                r"on {klass} with these indexers \[(0|1)\.0\] of "
                 "type float".format(klass=type(index).__name__)
             )
             with pytest.raises(TypeError, match=msg):
-                s[slice(-10.0, 10.0)]
+                s[l]
 
-            # getitem odd floats
-            for l, res in [
-                (slice(0.5, 1), slice(1, 2)),
-                (slice(0, 0.5), slice(0, 1)),
-                (slice(0.5, 1.5), slice(1, 2)),
-            ]:
+        # getitem out-of-bounds
+        for l in [slice(-10, 10), slice(-10.0, 10.0)]:
 
-                result = idxr(s)[l]
-                self.check(result, s, res, False)
+            result = s.loc[l]
+            self.check(result, s, slice(-10, 10), True)
 
-                # positional indexing
-                msg = (
-                    "cannot do slice indexing "
-                    r"on {klass} with these indexers \[0\.5\] of "
-                    "type float".format(klass=type(index).__name__)
-                )
-                with pytest.raises(TypeError, match=msg):
-                    s[l]
+        # positional indexing
+        msg = (
+            "cannot do slice indexing "
+            r"on {klass} with these indexers \[-10\.0\] of "
+            "type float".format(klass=type(index).__name__)
+        )
+        with pytest.raises(TypeError, match=msg):
+            s[slice(-10.0, 10.0)]
 
-            # setitem
-            for l in [slice(3.0, 4), slice(3, 4.0), slice(3.0, 4.0)]:
+        # getitem odd floats
+        for l, res in [
+            (slice(0.5, 1), slice(1, 2)),
+            (slice(0, 0.5), slice(0, 1)),
+            (slice(0.5, 1.5), slice(1, 2)),
+        ]:
 
-                sc = s.copy()
-                idxr(sc)[l] = 0
-                result = idxr(sc)[l].values.ravel()
-                assert (result == 0).all()
+            result = s.loc[l]
+            self.check(result, s, res, False)
 
-                # positional indexing
-                msg = (
-                    "cannot do slice indexing "
-                    r"on {klass} with these indexers \[(3|4)\.0\] of "
-                    "type float".format(klass=type(index).__name__)
-                )
-                with pytest.raises(TypeError, match=msg):
-                    s[l] = 0
+            # positional indexing
+            msg = (
+                "cannot do slice indexing "
+                r"on {klass} with these indexers \[0\.5\] of "
+                "type float".format(klass=type(index).__name__)
+            )
+            with pytest.raises(TypeError, match=msg):
+                s[l]
 
-        f(lambda x: x.loc)
+        # setitem
+        for l in [slice(3.0, 4), slice(3, 4.0), slice(3.0, 4.0)]:
```

Comment:
Possible to parametrize this since done in other tests? Maybe need to split off?

Reply:
this will need to be split off in a couple of places, will do in a separate pass